qiskit_ibm_provider/ibm_backend_service.py

```diff
@@ -682,13 +682,24 @@ def _create_backend_obj(
             instance: the current h/g/p.
         Returns:
             A backend object.
+        Raises:
+            QiskitBackendNotFoundError: if the backend is not in the hgp passed in.
         """
         if not instance:
             for hgp in hgps:
                 if config.backend_name in hgp.backends:
                     instance = to_instance_format(hgp._hub, hgp._group, hgp._project)
                     break
```

Comment:
In a similar note, we should raise if the backend is not found in any of the hgps when instance is not passed in